src/NumericalAlgorithms/FiniteDifference/PositivityPreservingAdaptiveOrder.hpp

```diff
@@ -0,0 +1,152 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <utility>
+
+#include "NumericalAlgorithms/FiniteDifference/FallbackReconstructorType.hpp"
+#include "NumericalAlgorithms/FiniteDifference/Reconstruct.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Math.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim>
+class Direction;
+template <size_t Dim>
+class Index;
+/// \endcond
+
+namespace fd::reconstruction {
+namespace detail {
+template <typename LowOrderReconstructor, bool PositivityPreserving>
```

Comment:
Rename `LowOrder~` to 'Fallback~', since I see `Fallback` is used a lot in the `.cpp` file?

(again, consistency of term used in code. I have no solid preference, but sticking to the term `Fallback` seems a good option :)

Reply:
I intentionally chose `LowOrder` because there's yet another fallback after the "fallback", and when you do 9,5,3,1 then 5,3,1 are all "fallback"s

Reply:
Okay!